src/main/java/de/hhu/propra/sharingplatform/service/UserService.java

```diff
@@ -148,5 +150,20 @@ private void validatePasswords(String password) {
         }
     }
 
+    private void validateAccountName(User user) {
```

Comment:
You can ask the user repo for all accountnames that match user.getAccountname(). If the return is not null, the username already exists. This will be probaply quicker than going through the arraylist.